### ISSUE TITLE
-->> FREE FOR TALK 7 <<---

### ISSUE THREAD

**Issue by iperov** [iperov/DeepFaceLab#196]
torr / mega upd

== 03.21.2019 ==

OpenCL build: fixed, now works on most video cards again.

old SAE model will not work with this update.
Fixed bug when SAE can be collapsed during a time

Added option
Use CA weights? (y/n, ?:help skip: n ) :
Initialize network with 'Convolution Aware' weights. 
This may help to achieve a higher accuracy model, but consumes time at first run.

Extractor:
removed DLIB extractor
greatly increased accuracy of landmarks extraction, especially with S3FD detector, but speed of 2nd pass now slower.
From this point on, it is recommended to use only the S3FD detector.
before https://i.imgur.com/SPGeJCm.gif
after https://i.imgur.com/VmmAm8p.gif

Converter: added new option to choose type of mask for full-face models.

Mask mode: (1) learned, (2) dst, (3) FAN-prd, (4) FAN-dst (?) help. Default - 1 : 
Learned – Learned mask, if you choose option 'Learn mask' in model. The contours are fairly smooth, but can be wobbly.
Dst – raw mask from dst face, wobbly contours.
FAN-prd – mask from pretrained FAN model from predicted face. Very smooth not shaky countours.
FAN-dst – mask from pretrained FAN model from dst face. Very smooth not shaky countours.
Advantages of FAN mask: you can get a not wobbly shaky without learning it by model.
Disadvantage of FAN mask: may produce artifacts on the contours if the face is obstructed.

**Comment by iperov** [comment]
someone try FAN-dst

**Comment by andy-ger** [comment]
Sorry, can't upload previews because i'm working on a private project.
But the result looks very good to me.
Less flickering than default type (learned).
Only drawback: it's quite slow compared to type=learned.
But as you know, that isn't really a problem to me as this is a one time job.
I think i will stick with FAN-dst in the future.

Again, good job @iperov 👍

**Comment by keyutao9** [comment]
FAN-dst works well, but the conversion speed is only 1/5 before.

Are you really fixing the SAE crash?

**Comment by iperov** [comment]
@keyutao9 not for some buggy rtx cards.

**Comment by keyutao9** [comment]
The new version of SAE similarity has basically reached expectations
I found a problem
When the SAE is trained to turn off pixel loss and style, the LOSS value will be trained very low. I usually train to 0.02 at 50K-80K. DST-LOSS has even been trained to 0.002, but the details of the preview are still unclear. . Is the LOSS value too low, and the training does not change for a long time, causing a crash caused by an infinite loop? And I found that most of the crashes in training with a LOSS value below 0.02. There is basically no collapse in the early stages of training.
I still had several crashed the training last night.
Later I tried to add some material to the SRC dataset to increase the src-loss value.
Then continue training, currently training for 5 hours, has not collapsed.
Is it possible to avoid a crash by amplifying the LOSS value.
Without opening pixel loss and style, LOSS loss seems to have lost its meaning within 0.02. Because no matter how long the training is, the LOSS value will fluctuate between 0.01 and 0.02 and will hardly continue to fall, but when LOSS = 0.02, the preview is still very blurry. Continued training will become clear. When I trained 192 resolution in front, 50KIter can reduce the LOSS value to 0.02. But it was not clear until the 300K preview. During this time, the LOSS value did not change significantly.
The LOSS of the original DF currently seems to be at a minimum of 0.07.

**Comment by overtort** [comment]
To me, This version will not be installed if avast is running. If anyone can't install,please pause vaccine.

**Comment by iperov** [comment]
@keyutao9 
what you want by constantly reporting crashes crashes crashes crashes crashes crashes here?
I already changed SAE several times specially for you. Now it's like DF.
Most people continue to create new cool fakes using SAE in the reddit.
Best option to you is no more use SAE, and stop annoying me.

**Comment by keyutao9** [comment]
I asked one of my friends to use the 1070TI, he also has the same situation as me, when the LOSS is close to 0.01 or below 0.01. The model is very easy to collapse.

**Comment by iperov** [comment]
Best option to you is no more use SAE, and stop annoying me.

**Comment by iperov** [comment]
@Tesla32x have you any model collapse during training?

**Comment by keyutao9** [comment]
![76`TK@RVDA%N{$582627~GI](https://user-images.githubusercontent.com/42455295/54807914-a60c5e80-4cb9-11e9-8ee8-aceadebe47c7.png)
After adding new materials to the SRC data set, the SRC-LOSS value becomes larger, and the training time is 5 hours in the morning and 4 hours in the afternoon. 9 hours of training, no crashes.

**Comment by kcimit** [comment]
I've got 1 crash, it appears to me it occurs around saving event. Either model crashing and automatically saving or it crashes after the save.
I'm adding backup code in ModelBase just in case it happens again:
`    
def save(self):
        
        if Path(self.get_strpath_storage_for_file('summary.txt')).exists():
            backup_path = os.path.join(self.model_path, datetime.datetime.now().strftime("backup %m-%d %H_%M"))
            os.mkdir(backup_path)
            for filename in glob.glob(os.path.join(self.model_path, '*.*')):
                shutil.copy(filename, backup_path)
`

**Comment by keyutao9** [comment]
> I've got 1 crash, it appears to me it occurs around saving event. Either model crashing and automatically saving or it crashes after the save.
> I'm adding backup code in ModelBase just in case it happens again:
> `
> def save(self):
> 
> ```
>     if Path(self.get_strpath_storage_for_file('summary.txt')).exists():
>         backup_path = os.path.join(self.model_path, datetime.datetime.now().strftime("backup %m-%d %H_%M"))
>         os.mkdir(backup_path)
>         for filename in glob.glob(os.path.join(self.model_path, '*.*')):
>             shutil.copy(filename, backup_path)
> ```
> `

My crash is different from yours. Although a crash occurs before and after saving the model, this rarely happens. I will only appear once in a few months of training. The crash I said was that the LOSS value suddenly became very large and the preview image turned pure white. When LOSS is very low, I usually show about 0.01. I just consulted some friends who often use style learning and pixel loss. They seem to crash very little, because after opening pixel loss and style, LOSS is basically difficult to drop below 0.2.

**Comment by kcimit** [comment]
> The crash I said was that the LOSS value suddenly became very large and the preview image turned pure white. 

Same here.

**Comment by keyutao9** [comment]
> > The crash I said was that the LOSS value suddenly became very large and the preview image turned pure white.
> 
> Same here.

I added new materials, SRC-LOSS increased, I haven't crashed after training for 10 hours today. I will turn on pixel loss before going to bed at night, increase the LOSS value and see if I can avoid the crash.

**Comment by iperov** [comment]
@keyutao9 upload your workspace, I will start to train it on my comp.

**Comment by keyutao9** [comment]
> @keyutao9 upload your workspace, I will start to train it on my comp.

OK

**Comment by iperov** [comment]
@keyutao9 downloaded your workspace.

seriously only 147 images in src ?
of course model will be overfitted.

**Comment by iperov** [comment]
minimum is 1000. But I recommend 4000 of various light conditions.

**Comment by keyutao9** [comment]
Yes, with a small amount of SRC, you can converge faster, and I will crash like 1000SRC. If these materials are not enough for you to test, I will use 1000 materials to train LOSS to 0.02 and send the workspace to you.

**Comment by iperov** [comment]
I dont need workspace anymore
You overfitted the model, of course it was crashed.
I thought you are advanced deepfaker :-\

**Comment by iperov** [comment]
> with a small amount of SRC, you can converge faster

worst phrase in machine learning. 
Converge mean generalization, ability to find common features by the network.
With small amount of images you are forcing the network to memorize samples, instead of find common features.

**Comment by keyutao9** [comment]
No, because the recent 192 and 256 resolutions are just testing, just want to see the results as quickly as possible, so I chose as few SRC and DST as possible, usually I use the same SRC and DST to speed up the convergence, but Still crashing. Except for this test I used a normal number of SRCs. Because the calculation of 256 resolution is too big and too long. I have also sent you some Asian faces, you should know that this is my favorite STAR, the material is more than 1500. It was only temporarily reduced when testing. Maybe this example is not typical enough, so I retrain 1000SRC until it crashes. If I resume SRC number training no longer crashes, then it is my operation problem.

**Comment by iperov** [comment]
Most people have no problems.
Most problems goes from rtx users.
Stop mentioning crashes here.

**Comment by Lucretious** [comment]
@iperov : May i ask about your recommendation for 4000 pics with different light conditions. Does that mean, that it is useful to use a source folder, that is actually mixed of different videos of one actor? Till now, I always did the exact opposite. Search for one video with a scene with ideal lighning. Mostly interviews of the chosen actor. I thought, that difference in lightning made it harder for the network to find the common landmarks, or even result in a bad model.

Also another question. What is the benefit of the optimization modes 2 and 3? I read, that they make a bigger network. But what does that do exactly? Does this mean, that you get better quality after a fixed amount of iterations? Or do you get to the same quality of a model faster, in comparison to using mode 1?

**Comment by titanrw** [comment]
I also thought it was better to have SRC pictures of uniform lighting.  As face style should adapt lighting of DST to converted face.  I have combined faces from different videos, but I still look for shots that have consistent lighting.  I will admit that most of my src face sets don't have 4000 pics.  Usually 500 or so.  Still had good luck though.

Optimization modes allow bigger models (either NN size, or resolution, or a combination of both) to be trained in the same amount of vram.  For example, 256 res was impossible on 6 gig cards before.  Now you can do 256 res on a 6 gig card using optimization mode.  My experience is that it's always slower, but it'll let you train higher configs that weren't possible before.

**Comment by wasyleque** [comment]
Sae crashes on 1080Ti win 10 too in new build since 14.03.2019
<img src="https://zapodaj.net//images/74c45907b4a61.jpg"></img>

**Comment by titanrw** [comment]
Stupid question.  Isn't a BG style of 100 more likely to cause this, as it goes overboard trying to train in the background?  I've never had to go above BG style 10 or so.  I've definitely had to revert a model before because of turning up the styling too far.

**Comment by wasyleque** [comment]
I made many movies with bg 100 with superb effect (objects on face looks great, hairs on face looks perfect)

**Comment by Lucretious** [comment]
@titanrw  Thank you, I never would have guessed that. That means, that I used the optimization mode totally wrong... :-D
Right now I am trying to build a src set with about 3500 faces, made from the best quality faces of 5 videos, which really are very different lightning. Handpicked, curated faces. Let´s see how that will affect the neural network.

**Comment by titanrw** [comment]
@Lucretious - I made some comparisons on the optimizer mode, posted over on mrdeepfakes.com:

https://mrdeepfakes.com/forums/thread-questions-about-new-optimizer?pid=4240#pid4240

And

https://mrdeepfakes.com/forums/thread-questions-about-new-optimizer?pid=4254#pid4254

**Comment by Kenturin** [comment]
@wasyleque From my understanding, CA weights/initialization leads to higher model accuracy AND **faster** convergence.  Since convergence happens faster, it's possible that **overfitting** may occur faster as well.  With smaller batch sizes the weights are updated more frequently which may cause instability in the model/cause it to overfit faster (just a theory).  I did have a 1000+ set model 'crash' using CA, however, I was able to remedy it by reloading the most recent backup and increasing batch size for a bit until it stabilized.  It's still stable 50k iterations later.  TL;DR - Aside from increasing data set size, try increasing batch size for a time and see if that helps with stability.

@iperov  the FAN-dst mask and the new CA weights are really impressive!!  So far they're game changers!

**Comment by Lucretious** [comment]
@titanrw : Thanks for the links. These are useful to get a feeling for what is possible, at what cost. Now I finally can maybe utilize mode 2 to try working with 256 resolution.

**Comment by iperov** [comment]
@wasyleque 
> Sae crashes on 1080Ti win 10 x64 too in new build since 14.03.2019

easy to reach crash with style values.

**Comment by iperov** [comment]
>  Since convergence happens faster, it's possible that overfitting may occur faster as well

no. keyut overfitted his model with 149 images.

**Comment by Kenturin** [comment]
> > Since convergence happens faster, it's possible that overfitting may occur faster as well
> 
> no. keyut overfitted his model with 149 images.

I'm agreeing with you.  Small data set leads to overfitting.

**Comment by Kenturin** [comment]
> @wasyleque
> 
> > Sae crashes on 1080Ti win 10 x64 too in new build since 14.03.2019
> 
> easy to reach crash with style values.

Especially with BG style of 100

**Comment by kcimit** [comment]
maybe someone could explain it to me - actually, what's the point of learning background with bg style if at the end only face within established landmarks will be transferred?

**Comment by wasyleque** [comment]
@kcimit i think that hairs and objects on face looks more like in org dst without blurred almost. And bg around face looks more like org dst too. That's what i observed about it.

**Comment by overtort** [comment]
Can I train the two faces of the dst at the same time?
I've tried it, but it seems to work. Even if not perfect.

**Comment by iperov** [comment]
new suspicion that sae collapse was affected by the initialization of the weights by random_normal(0, 0.2).
Classic models have default uniform initialization.
Also CA weights also can produce collapse too.

**Comment by keyutao9** [comment]
Retrain or continue training？

**Comment by iperov** [comment]
@keyutao9 check with
1) last commit
2) last commit + ```Adam(clipnorm=1.0, ... )```

**Comment by keyutao9** [comment]
Adam (clipnorm=1.0 modified, I trained for 6 hours, no crash, but the LOSS value does not seem to fall or the decline is very slow, because I barely see any changes
I test the last commit now

**Comment by iperov** [comment]
@keyutao9 without CA weights of course.

**Comment by keyutao9** [comment]
I know

**Comment by keyutao9** [comment]
![`TI%2G7)A_ KJ(_QR{A5187](https://user-images.githubusercontent.com/42455295/54866223-30cd8600-4dac-11e9-83fa-85a5b80ac567.png)



last commit also crashed.But automatically recover within 20 minutes。Continue training

**Comment by iperov** [comment]
@keyutao9 again you doing something wrong?

you wrote

> I test the last commit now

2 hours ago. But in your log 78229 * 1.3s = 28 hours.

**Comment by keyutao9** [comment]
I didn't recreate the model, but I just updated the latest source code. I continue my last training.
I am testing whether it will crash, because the crash will only happen when the LOSS value is very low, and training takes at least 2 days to train from 0 to crash.
At the moment, the new fix seems to be very useful, and even if the LOSS value suddenly rises, it will recover in a short time. Compared with Adam (clipnorm=1.), the LOSS decline is very stable, and there will be no drop.
Adam (clipnorm=1. will cause the LOSS value to drop very slowly.
It looks good now. If I train to not crash tomorrow morning, maybe the crash is really fixed.

**Comment by iperov** [comment]
@keyutao9

> I didn't recreate the model, but I just updated the latest source code. I continue my last training.

 and this is wrong. Change in last commit  is set weights initialization to default. So you have to retrain model.

**Comment by keyutao9** [comment]
OK

**Comment by iperov** [comment]
I implemented ReflectionPadding2D for plaidML

new option with it will remove gray line from border

![Uploading python_2019-03-23_22-22-14.jpg…]()

**Comment by iperov** [comment]
also I got model crash with CA weights.
It seems any weights that differents from default 'glorot_uniform' will cause model crash.
21 feb SAE got new weights initialization, perhaps model crash has begun on this date.

**Comment by FailFake** [comment]
> also I got model crash with CA weights.
> It seems any weights that differents from default 'glorot_uniform' will cause model crash.
> 21 feb SAE got new weights initialization, perhaps model crash has begun on this date.

I got my first ever model crash @ iteration 40K w/latest release, CA on. SAE DF, PL off, styling off.

**Comment by titanrw** [comment]
I started a new model from scratch using the mega distribution from 03/21.  Set it to 192 res, 576x44 dims, df, CA weights on.  No need for 256 on this one, so I upped dims a little.

Trained good to about 35k.  I stopped, and raised batch size from 6 to 16.  Then at 40k it crashed.  Pure white / yellow previews.  I suspect this was because I still had pixel loss off, even though loss was already quite good.  Reverted model to 10 mins before crash, and continued train with pixel loss on.

No problem since.  For only 50k iter, the previews look really good.  This definitely seems to converge faster than other versions.

This has lots of faces.  2,000 src, and 10,000 dst.

**Comment by wasyleque** [comment]
@kcimit where to paste that code ? does not work, cause I dont know python :(

**Comment by iperov** [comment]
@titanrw I will remove ca weights.

**Comment by titanrw** [comment]
I'm happy with CA weights.  It trained very fast.  In my one example you just have to pay attention to when pixel loss should turn on.  Once PL was on, no further problems.

**Comment by iperov** [comment]
nope, I got crash even with pixel loss.
Default initialization is the best.

**Comment by keyutao9** [comment]
New converter, even if you don't use the FAN pre-training model,
The learned and DST mask mode conversion speed is also very slow, about 1/8 of the previous
is this normal?
![FRC 38KXO)GV0)7K}N5UY$D](https://user-images.githubusercontent.com/42455295/54875349-61f39800-4e38-11e9-9beb-6d56aa9f7ad7.png)

**Comment by iperov** [comment]
all fine for me
![2019-03-24_09-31-23](https://user-images.githubusercontent.com/8076202/54875387-9acf4500-4e17-11e9-9428-77795ae4c023.gif)

**Comment by iperov** [comment]
before posting such statements, please compare with previous build from mega.nz.

**Comment by keyutao9** [comment]
I  re-downloaded the previous version of MEGA.
I couldn't convert it . Later I found that the avatar extracted from the new version could not be used in the previous version, so I used the 11.3.2019CUDA10 version to re-extract and convert the video with the same resolution.
![5H 8`$K%8XHIPE6JX%`6ULD](https://user-images.githubusercontent.com/42455295/54875981-1eeaf200-4e43-11e9-8cf2-529cdbfec210.png)

**Comment by iperov** [comment]
then dont use new versions.

**Comment by iperov** [comment]
I decided to remove VG archi
but I found that residual blocks in decoder can help to acquire higher details.

**Comment by Lucretious** [comment]
@keyutao9: I actually have the same problem. I am using the lastest version, and all modes of the newest converter are very slow. For me they even seem to have the same speed (~2,4 it/s). I will try an older version to see if the problem still exists.

**Comment by iperov** [comment]
just dont use DFL if you have any problems

**Comment by iperov** [comment]
I unsubscribed from this thread. Flood anything you want.

**Comment by wasyleque** [comment]
Newest build on 2 different PC without any problems 53k iters and 64k iters.
```
== Model options:
== |== batch_size : 16
== |== sort_by_yaw : True
== |== random_flip : False
== |== resolution : 128
== |== face_type : f
== |== learn_mask : True
== |== optimizer_mode : 1
== |== archi : df
== |== ae_dims : 698
== |== e_ch_dims : 54
== |== d_ch_dims : 27
== |== d_residual_blocks : False
== |== remove_gray_border : False
== |== multiscale_decoder : True
== |== pixel_loss : True
== |== face_style_power : 0.1
== |== bg_style_power : 4.0
== Running on:
== |== [0 : GeForce GTX 1080 Ti]
```

**Comment by FailFake** [comment]
[https://www.nvidia.com/research/inpainting/selection](url) (Demo)
[https://github.com/NVIDIA/partialconv](url) (Official Github)

Has anyone tried this for fixing obstructions like eyeglasses etc? As a layperson I'd love a standalone/download version as easily usable as DFL, but don't see one available yet.

Wary of uploading any of my SRC images to the demo site despite many of them would benefit from having eyeglasses removed non-destructively. I don't have much SRC to work with so I need to improve what I do have. I can't seem to get any usable results out of Photoshop's content-aware tools for removing SRC obstructions, just not skilled enough.

**Comment by andy-ger** [comment]
That's cool!

![image](https://user-images.githubusercontent.com/47699661/55034611-1dbdee80-5016-11e9-92ff-7153598f8c16.png)

**Comment by oracle9i88** [comment]
cool

**Comment by Kenturin** [comment]
I'm sure some have seen this, but I thought I'd share.  SRGAN is promising as an additional tool in workflow.  I've had some success with lower resolution fakes.  I've used SRGAN to upscale videos/pictures, then run them through DFL for even better results in some cases.  This isn't something for everyday faking workflow...it's just useful if you can't find a high quality src or dst.  However, it obviously doesn't work if your 'low resolution' video or pictures are already blurry/grainy/shit-quality
...as always, garbage in = garbage out.

TL;DR - SRGAN takes low resolution images and upscales them while maintaining/recovering high-frequency details.

Links are below for any interested.  @iperov @andy-ger I thought you'd like it, cheers!

https://github.com/krasserm/super-resolution

https://github.com/tensorlayer/srgan

Original Paper: https://arxiv.org/abs/1609.04802

**Comment by Kenturin** [comment]
EDIT:  To clarify, the left image started as a 128 x 128 image of N.Cage.  I ran it through an SRGAN script that upscaled it 4x to 512 x 512.  I ran the DFL SF3D extractor on both which outputs at 256 x 256 and these were the results.  Both images used are 256 x 256 but the SRGAN 'enhanced' one is very good quality.

**Comment by tfis2** [comment]
Crash, when converting with alpha mask:

File "E:\DeepFaceLabCUDA9.2SSE_built 2603\_internal\DeepFaceLab\mainscripts\Converter.py", line 110, in process_data
    image = self.converter.convert_face(image, image_landmarks, self.debug)
  File "E:\DeepFaceLabCUDA9.2SSE_built 2603\_internal\DeepFaceLab\converters\ConverterMasked.py", line 333, in convert_face
    out_img = np.clip( img_bgr*(1-img_mask_blurry_aaa) + (out_img*img_mask_blurry_aaa) , 0, 1.0 )
ValueError: operands could not be broadcast together with shapes (720,1280,4) (720,1280,3)

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "E:\DeepFaceLabCUDA9.2SSE_built 2603\_internal\DeepFaceLab\joblib\SubprocessorBase.py", line 68, in _subprocess_run
    result = self.process_data (data)
  File "E:\DeepFaceLabCUDA9.2SSE_built 2603\_internal\DeepFaceLab\mainscripts\Converter.py", line 117, in process_data
    raise Exception( 'Error while converting face_num [%d] in file [%s]: %s' % (face_num, filename_path, e_str) )
Exception: Error while converting face_num [1] in file [E:\DeepFaceLabCUDA9.2SSE_built 2603\workspace\data_dst\04140.jpg]: Traceback (most recent call last):
  File "E:\DeepFaceLabCUDA9.2SSE_built 2603\_internal\DeepFaceLab\mainscripts\Converter.py", line 110, in process_data
    image = self.converter.convert_face(image, image_landmarks, self.debug)
  File "E:\DeepFaceLabCUDA9.2SSE_built 2603\_internal\DeepFaceLab\converters\ConverterMasked.py", line 333, in convert_face
    out_img = np.clip( img_bgr*(1-img_mask_blurry_aaa) + (out_img*img_mask_blurry_aaa) , 0, 1.0 )
ValueError: operands could not be broadcast together with shapes (720,1280,4) (720,1280,3)

Error message come several times, then conversion stopped.

**Comment by wasyleque** [comment]
check If You dont have two same faces with *_0.jpg and *_1.jpg, for make sure move *_1.jpg files to another folder before conversion.

**Comment by YouYouExcellent** [comment]
the preview windows won't pop out, on the latest codes.

**Comment by wasyleque** [comment]
@HirezmingD there will be no answer, You need to choose which model You like, author is tired of answering stupid questions (no offense) so he`s focusing on the code and real issues for now.

**Comment by chervonij** [comment]
@HirezmingD 
H128 loss and SAE loss not equal eachother
Evaluate the quality of training only by preview

**Comment by HirezmingD** [comment]
@chervonij @wasyleque Thank you !

**Comment by tfis2** [comment]
> 
> 
> @tfis2 check If You dont have two same faces with *_0.jpg and *_1.jpg, for make sure move *_1.jpg files to another folder before conversion.

Thanks. 
Never had a problem with *_1 files.
Nevertheless the frame # mentioned in the error message was in a series of _0 frames.

**Comment by wasyleque** [comment]
@tfis2 but it helped ?

**Comment by tfis2** [comment]
> 
> 
> @tfis2 but it helped ?

I don't know.... 
Tried the last build (I thought "Converter: fix for input images with ch > 3" could solve the problme) and now conversion crashes right from the beginning...

**Comment by wasyleque** [comment]
@tfis2
For me an error:
Exception: Error while converting face_num [1] 
Means face from file *_1*, do what I suggest IT should works. Find files with that  pattern in Explorer and move them to another folder.

**Comment by tfis2** [comment]
> 
> 
> @tfis2
> For me an error:
> Exception: Error while converting face_num [1]
> Means face from file __1_,do what I suggest IT should works. Find files with that pattern in Explorer and move them to another folder.

I understand. But the error occurs only if using alpha mask (4th channel)
The actual update lets the conversion process crash immediately after collecting the alignements.
But only with Cuda9 version.
With Cuda10 conversion works till frame 4140 (after update). I'll try converting the _0 and the _1 files separately and merge the folders, like you mentioned. 

BTW: It seems, that "super resolution" does nothing at all, but I'm still on LIAEF. Maybe with SAE it works better, but i have to retrain the model, first.

**Comment by chervonij** [comment]
@tfis2 
Don't spent your time.
Super resolution not good in SAE too.
If you want a better result, try Residual Block in SAE

**Comment by iperov** [comment]
I ported https://github.com/jiny2001/dcscn-super-resolution which is better than srgan.
it seems there is nothing to superres if no information provided like a predicted face.
superres cannot imagine an eye if there is no eye in image.
But edges can be more clear now, and works better than same amount of regular 'sharpen'
![jJA71Vy 1](https://user-images.githubusercontent.com/8076202/55211155-4c012100-5204-11e9-95b9-51c7bef48e03.gif)

**Comment by FailFake** [comment]
Could be I just had a perfect DST the first time I used it, but I seem to be getting way more false positives extracting faces with S3FD lately compared to when first implemented - as many false now as MT did (where the match is nothing resembling a face, usually when no faces at all are in the scene). Just wondering if sensitivity was increased or something?

**Comment by gessyoo** [comment]
@iperov, very good work, I was experimenting with super-resolution to try to improve the converted face.

**Comment by q1q2w3r** [comment]
I just want to grab all the faces myself, including every point of the facial features. Is there any way to achieve this? This is a must-have path for obscuring the face.

**Comment by waterfish1010** [comment]
> the preview windows won't pop out, on the latest codes.

same problem, no preview window since 03.26 version

**Comment by iperov** [comment]
I thought I unsubscribed from this whining.
But if someone mention me, I subscribed again
